lib/src/main/java/graphql/nadel/engine/transform/NadelRenameArgumentInputTypesTransform.kt

```diff
@@ -36,20 +34,11 @@ internal class NadelRenameArgumentInputTypesTransform : NadelTransform<State> {
         overallField: ExecutableNormalizedField,
         hydrationDetails: ServiceExecutionHydrationDetails?,
     ): State? {
-
-        var changeCount = 0
-        val newFieldArgs: Map<String, NormalizedInputValue> = overallField.normalizedArguments
-            .mapValues { (_, inputValue) ->
-                val newInputValue = renameInputValueType(inputValue, executionBlueprint, service)
-                if (newInputValue != inputValue) {
-                    changeCount++
-                }
-                newInputValue
-            }
-        return if (changeCount == 0) {
-            null
-        } else {
-            State(newFieldArgs)
+        // Transform if there's any arguments at all
+        // todo: this won't account for cases where a transform before this injected new arguments…
+        // But that's not a big deal right now anyway…
+        return State.takeIf {
+            overallField.normalizedArguments.isNotEmpty()
         }
```

Comment:
See this whats pisses me off with Kotlin and a programming aim to be write as LEAST code as possible

I am not 100% sure how to read it and its not clearer - it may be concise BUT its not clearer code.

`return overallField.normalizedArguments.isNotEmpty() ? State : null` - look its less lines and IMO clearer

Reply:
Maybe of there is was a decent amount if code in lambda then a `takeIf` might be worth it - but there isnt.

I find it Yoda coding

```
return DarkSide.takeIf { angerLeadsToHate }
```

Reply:
Yup fair, I will fix that lol.

Reply:
Just another 2 cents, some of the default Kotlin extension functions like `let` `also` `takeIf` etc. are kind of like `map` `onEach` `filter` etc. It's just some functions you have to learn. It's part of the language. Just like how you had to learn those transforms as part of learning Reactive programming.

Sometimes like in this case it's useless, but there's also valid use cases like

```kotlin
listOf()
    .filter { matchesPredicate(it) }
    .first()
    .let { it.string }
    .takeIf { it.isValid }
```

etc.

Reply:
Sure I get it - takeIf is a scope function (or whatever they are called) and in the case above - perfect.

But for a simple case (not a stream of values) then `return list.takeIf { thing.isNotEmpty }` is less clear than it could  be